qcodes/instrument_drivers/tektronix/Keithley_2400.py

```diff
@@ -77,11 +77,30 @@ def __init__(self, name, address, **kwargs):
                            label='Current integration time')
 
         self.add_parameter('resistance',
-                           get_cmd=':READ?',
+                           get_cmd=self._get_read_output_protected,
                            get_parser=self._resistance_parser,
                            label='Resistance',
                            unit='Ohm')
 
+    def _get_read_output_protected(self) -> str:
+        """
+        This wrapper function around ":READ?" exists because calling
+        ":READ?" on an instrument with output disabled is an error.
+        So first we check that output is on and if not we return
+        nan for volt, curr etc.
+        """
+        output = self.output.get_latest()
+        if output is None:
+            # if get_latest returns None we have
+            # to ask the instrument for the status of output
+            output = self.output.get()
+
+        if output == 1:
+            msg = self.ask(':READ?')
+        else:
+            msg = "nan,nan,nan,nan,nan"
```

Comment:
Why not raise an exception here?

Reply:
Yes I thought about that. My main motivation for not doing that was to prevent adding it to the station from triggering an exception

Reply:
My worry is that if you add it to a station and do some measurements you might miss the fact that it's not actually recording any data. I don't think there's a perfect solution, though I would prefer an exception.

Reply:
@jenshnielsen, sorry, I did not understand what you meant. Despite that I tend to agree with @spxtr.

Do you think it is enough to add a note to the driver docstring (or docstring of each affected parameter) that "output" should be on before reading values, otherwise an exception will be thrown?

Reply:
To be more clear. I think it's fine for the get to throw when called directly, and yes documenting is a good idea. 

I was more concerned with the automatic get that happens at setup time but the snapshot will catch that exception anyway so either way should be fine.